cores/esp8266/pgmspace.h

```diff
@@ -112,8 +112,13 @@ static inline uint16_t pgm_read_word_inlined(const void* addr) {
 }
 
 // Make sure, that libraries checking existence of this macro are not failing
+#ifdef __PROG_TYPES_COMPAT__
```

Comment:
Names starting with double underscore are usually reserved for the compiler and standard library, can we use something like ESP8266_PROGMEM_TYPES_COMPAT?